### ISSUE TITLE
Loading props as Autocomplete MUI

### ISSUE THREAD

**Issue by Maxim-BC** [mikepricedev/mui-tree-select#7]
Hi!
I have a question.
Is it possible in the latest version of mui-tree-select to pass the loading, noOptionsText, loadingText properties, as in Autocomplete MUI? 
I pass these props, but they don't work in mui-tree-select.

Thank you in advance!

**Comment by mikepricedev** [comment]
Hi @Maxim-BC , 

[`noOptionsText`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#noOptionsText), and [`loadingText`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#loadingText) do pass through.

`loading` however, is a little different.  Because `TreeSelect` utilizes callbacks instead of the `options` prop  to retrieve options, those callbacks can be async.    The `loading` prop state is detected base on the return type of those methods and  [passed](https://github.com/mikepricedev/mui-tree-select/blob/f3871c7cf548f998544c56a29da7a7dda5e9bb0c/src/TreeSelect.tsx#L696) to the underlying `Autocomplete`.   That is to say it is handled internally.

`loading === true` whenever [`getParent`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#getParent) or [`getChildren`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#getChildren) return a `Promise` [while looking up](https://github.com/mikepricedev/mui-tree-select/blob/f3871c7cf548f998544c56a29da7a7dda5e9bb0c/src/useTreeSelect.ts#L1400) the current branch's options (And while looking up the current branch's path).

That said is there a scenario where you would want to trigger the loading state that is not related to asynchronously fetching branch options?

Respectfully,
M Price

**Comment by mikepricedev** [comment]
While putting together this [sandbox](https://codesandbox.io/s/unruffled-grass-1fbwvv) to demo [noOptionsText](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#noOptionsText), and [loadingText](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#loadingText) pass through.  I noticed that it is not passing through the override at the root branch level.

Thanks for letting me know!

**Comment by Maxim-BC** [comment]
> Привет@Maxim-BC,
> 
> [`noOptionsText`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#noOptionsText), и [`loadingText`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#loadingText)пройти через.
> 
> `loading`однако немного отличается. Поскольку для получения параметров `TreeSelect`используются обратные вызовы вместо `options`реквизита, эти обратные вызовы могут быть асинхронными. Состояние `loading`свойства определяется на основе типа возвращаемого значения этих методов и [передается](https://github.com/mikepricedev/mui-tree-select/blob/f3871c7cf548f998544c56a29da7a7dda5e9bb0c/src/TreeSelect.tsx#L696) базовому объекту `Autocomplete`. То есть обрабатывается внутри.
> 
> `loading === true`всякий раз [`getParent`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#getParent)или [`getChildren`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#getChildren)возвращать `Promise` [при поиске](https://github.com/mikepricedev/mui-tree-select/blob/f3871c7cf548f998544c56a29da7a7dda5e9bb0c/src/useTreeSelect.ts#L1400) параметров текущей ветки (и при поиске пути к текущей ветке).
> 
> Тем не менее, есть ли сценарий, в котором вы хотели бы инициировать состояние загрузки, не связанное с асинхронной выборкой параметров ветвления?
> 
> С уважением, М Прайс



> Hi @Maxim-BC ,
> 
> [`noOptionsText`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#noOptionsText), and [`loadingText`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#loadingText) do pass through.
> 
> `loading` however, is a little different. Because `TreeSelect` utilizes callbacks instead of the `options` prop to retrieve options, those callbacks can be async. The `loading` prop state is detected base on the return type of those methods and [passed](https://github.com/mikepricedev/mui-tree-select/blob/f3871c7cf548f998544c56a29da7a7dda5e9bb0c/src/TreeSelect.tsx#L696) to the underlying `Autocomplete`. That is to say it is handled internally.
> 
> `loading === true` whenever [`getParent`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#getParent) or [`getChildren`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#getChildren) return a `Promise` [while looking up](https://github.com/mikepricedev/mui-tree-select/blob/f3871c7cf548f998544c56a29da7a7dda5e9bb0c/src/useTreeSelect.ts#L1400) the current branch's options (And while looking up the current branch's path).
> 
> That said is there a scenario where you would want to trigger the loading state that is not related to asynchronously fetching branch options?
> 
> Respectfully, M Price

Hello @M Price,

I have several mui-tree-select that don't have options initially, but when "onFocus" is triggered, a function is launched that loads options, I would like to have a message about loading options when loading options, not "No options". 

Respectfully,
Maxim

**Comment by Maxim-BC** [comment]
> Hi @Maxim-BC ,
> 
> [`noOptionsText`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#noOptionsText), and [`loadingText`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#loadingText) do pass through.
> 
> `loading` however, is a little different. Because `TreeSelect` utilizes callbacks instead of the `options` prop to retrieve options, those callbacks can be async. The `loading` prop state is detected base on the return type of those methods and [passed](https://github.com/mikepricedev/mui-tree-select/blob/f3871c7cf548f998544c56a29da7a7dda5e9bb0c/src/TreeSelect.tsx#L696) to the underlying `Autocomplete`. That is to say it is handled internally.
> 
> `loading === true` whenever [`getParent`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#getParent) or [`getChildren`](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#getChildren) return a `Promise` [while looking up](https://github.com/mikepricedev/mui-tree-select/blob/f3871c7cf548f998544c56a29da7a7dda5e9bb0c/src/useTreeSelect.ts#L1400) the current branch's options (And while looking up the current branch's path).
> 
> That said is there a scenario where you would want to trigger the loading state that is not related to asynchronously fetching branch options?
> 
> Respectfully, M Price

Hello M Price,

I have several mui-tree-select that don't have options initially, but when "onFocus" is triggered, a function is launched that loads options, I would like to have a message about loading options when loading options, not "No options". 

Respectfully,
Maxim

**Comment by mikepricedev** [comment]
Gotcha,

Would something like this work?

```js
const [focused, setFocused] = useState(false);

const noResolve = useMemo(() => new Promise((_) => {}), []);

const getChildren = useCallback((node)=>{

  if(!focused) {
    return noResolve;
  }
  
  // ... load options

}, [focused, noResolve])

const onFocus = useCallback(()=>setFocused(true), []);

```

[Example](https://codesandbox.io/s/cranky-grothendieck-4ykhvl?file=/index.js)

**Comment by mikepricedev** [comment]
> While putting together this [sandbox](https://codesandbox.io/s/unruffled-grass-1fbwvv) to demo [noOptionsText](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#noOptionsText), and [loadingText](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#loadingText) pass through. I noticed that it is not passing through the override at the root branch level.
> 
> Thanks for letting me know!

Fixed with 6a52d241285411622f858592649a980ce67f25fe

**Comment by Maxim-BC** [comment]
> > While putting together this [sandbox](https://codesandbox.io/s/unruffled-grass-1fbwvv) to demo [noOptionsText](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#noOptionsText), and [loadingText](https://mikepricedev.github.io/mui-tree-select/interfaces/TreeSelectProps.html#loadingText) pass through. I noticed that it is not passing through the override at the root branch level.
> > Thanks for letting me know!
> 
> Fixed with [6a52d24](https://github.com/mikepricedev/mui-tree-select/commit/6a52d241285411622f858592649a980ce67f25fe)

Thank you very much! Now everything is working as it should.
